### ISSUE TITLE
ccxt.pro does not support proxy with Binance

### ISSUE THREAD

**Issue by williamvgam** [ccxt/ccxt#20315]
### Operating System

window

### Programming Languages

Python

### CCXT Version

4.1.14

### Description

ccxt.pro does not support proxy with Exchanges( including: Binance,Bybit,Bitmart,Huobi)
Exchenges ( okx, gate) are good with proxy

### Code

import ccxt.pro
from asyncio import run

print(ccxt.__version__)

exchange = ccxt.pro.binance({
    'options': {
        'defaultType': 'swap',
    },
    'newUpdates': False,
})
exchange.socksProxy = 'socks5://127.0.0.1:1008/'

await exchange.watch_order_book('BTC/USDT:USDT')

```
RequestTimeout                            Traceback (most recent call last)
~\AppData\Local\Temp/ipykernel_16900/3865262703.py in <module>
     12 exchange.socksProxy = 'socks5://127.0.0.1:1008/'
     13 
---> 14 await exchange.watch_order_book('BTC/USDT:USDT')

~\anaconda3\lib\site-packages\ccxt\pro\binance.py in watch_order_book(self, symbol, limit, params)
    215         message = self.extend(request, params)
    216         # 1. Open a stream to wss://stream.binance.com:9443/ws/bnbbtc@depth.
--> 217         orderbook = await self.watch(url, messageHash, message, messageHash, subscription)
    218         return orderbook.limit()
    219 

RequestTimeout: Connection timeout  

```

**Comment by carlosmiei** [comment]
@williamvgam Thanks for reporting it, we will apply a patch shortly.

**Comment by ttodua** [comment]
@williamvgam I forgot to reply here, yes, the patch was made and it was merged. please update your ccxt package and let us know if you face any issues still.

**Comment by carlosmiei** [comment]
@ttodua @williamvgam We didn't release it, but we will shortly therefore this patch should be available in the next version, 4.1.86

**Comment by williamvgam** [comment]
@ttodua @carlosmiei That is cool, thank you all!
